.github/workflows/codeql.yaml

```diff
@@ -0,0 +1,35 @@
+# Ref: https://tinyurl.com/5d6rr8s7
+name: "CodeQL"
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ["javascript"]
```

Comment:
I'm curious if we wanna include typescript here too since we may have some problems with types that can be caught by codeql. I know that TypeScript compiles down to Javascript but I think there may be some value in adding that to the language matrix.

Reply:
https://github.com/github/codeql-action/issues/365